### PR TITLE
Create markers.lua

### PR DESCRIPTION
**Pull request by fusion-develops**
For ESX point system simpler version of ox libs

### CODE REVIEW

**Comment by Gellipapa** [comment]
Hi! @fusion-develops 

I like your solution and thanks for pr, but I don't know how happy Linden would be to use exactly the same name, so he's really referring to the ox_lib points function. 

If we don't look at this part because we are, say, just drawing inspiration from ox_lib, then we have name convention problems and probably the code doesn't work because it is used somewhere "inside" or "Inside" something in code.

Have you tested this code?

**Comment by fusion-develops** [comment]
> Hi! @fusion-develops
> 
> I like your solution and thanks for pr, but I don't know how happy Linden would be to use exactly the same name, so he's really referring to the ox_lib points function.
> 
> If we don't look at this part because we are, say, just drawing inspiration from ox_lib, then we have name convention problems and probably the code doesn't work because it is used somewhere "inside" or "Inside" something in code.
> 
> Have you tested this code?

I've redone it and fixed the bugs and changed the name to Markers since that's what it would mostly be used for once approved I can use this to redo all the add-on resources

File: [core]/es_extended/client/modules/points.lua [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+end
+
+CreateThread(function()
+    while true do 
+        Wait(0)
+        local InPoint = false
+        local Coords = GetEntityCoords(ESX.PlayerData.Ped)
+        for i = 1, #Markers do
+            local Marker = Markers[i]
+            Marker.distance = #(Coords - Marker.Coords)
```
Comment (Gellipapa) [inline [core]/es_extended/client/modules/points.lua:17]: Marker.Coords doesn't exist, and you should handle vector3. I mean convert if not vector 3.

File: [core]/es_extended/client/modules/points.lua [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+CreateThread(function()
+    while true do 
+        Wait(0)
+        local InPoint = false
+        local Coords = GetEntityCoords(ESX.PlayerData.Ped)
+        for i = 1, #Markers do
+            local Marker = Markers[i]
+            Marker.distance = #(Coords - Marker.Coords) 
+            if Marker.distance < Marker.radius then 
+                if Marker.status and Marker.Inside then
```
Comment (Gellipapa) [inline [core]/es_extended/client/modules/points.lua:19]: Marker.Inside, Marker.inside bad variable name, and don't use. You dont't set Marker.inside.

File: [core]/es_extended/client/modules/points.lua [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+function ESX.CreateMarker(data) 
+    local Index = #Points+1
+    Markers[Index] = data
+    return Markers[Index]
+end
+
+CreateThread(function()
+    while true do 
+        Wait(0)
+        local InPoint = false
```
Comment (Gellipapa) [inline [core]/es_extended/client/modules/points.lua:13]: Change all variable is correct lowerCamelCase. Thanks

**Review by Gellipapa — CHANGES_REQUESTED** [review]
Please test your code before open PR. Thanks a lot.

File: [core]/es_extended/client/modules/points.lua (hunk 1 above)
Comment (fusion-develops) [reply [core]/es_extended/client/modules/points.lua]: Done

File: [core]/es_extended/client/modules/points.lua (hunk 2 above)
Comment (fusion-develops) [reply [core]/es_extended/client/modules/points.lua]: Fixed Marker.Inside is now a check if the function exists

File: [core]/es_extended/client/modules/points.lua (hunk 3 above)
Comment (fusion-develops) [reply [core]/es_extended/client/modules/points.lua]: Removed Variable for a better way to do what it was meant for

File: [core]/es_extended/client/modules/markers.lua [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    while true do 
+        Wait(1500)
+        local Coords = GetEntityCoords(ESX.PlayerData.Ped)
+        for i = 1, #Markers do
+            local Marker = Markers[i]
+            Marker.distance = #(Coords - Marker.coords) 
+            if Marker.distance < Marker.radius then 
+                if Marker.status and Marker.Inside and not Marker.running then 
+                    Marker.running = true
+                    Markers:Run(i)
```
Comment (Gellipapa) [inline [core]/es_extended/client/modules/markers.lua:67]: hm, if i have 5 markers i created 5 thread?  I think we should avoid this and run code once thread.

**Review by Gellipapa — CHANGES_REQUESTED** [review]

File: [core]/es_extended/client/modules/markers.lua (hunk 4 above)
Comment (Gellipapa) [reply [core]/es_extended/client/modules/markers.lua:67]: and please change all variable is correct lowerCamelCase. Thanks

Comment (fusion-develops) [reply [core]/es_extended/client/modules/markers.lua]: Fixed both brother

**Review by Gellipapa — COMMENTED** [review]
Do you tested this refactored code? Probably the while loop will cause you to stop the rest of the code from running if there is any.

**Comment by Gellipapa** [comment]
Closed due to inactivity.

**Pull request closed without merging**